### ISSUE TITLE
Support for Autoprovision

### ISSUE THREAD

**Issue by justindavies** [Azure/AKS#2442]

**Comment by PixelRobots** [comment]
Anymore information about this. The title sounds interesting.

**Comment by clarenceb** [comment]
Can we get a 1-liner on what this is?

**Comment by mehrdadsilver** [comment]
What is the difference between this and AKS Karpenter https://github.com/Azure/karpenter ?

**Comment by stevehipwell** [comment]
@kasunsjc no I didn't.

**Comment by justindavies** [comment]
We are in the process of publishing the documentation for the feature.

**Comment by wdarko1** [comment]
@

> @stevehipwell Did you find the documentation for this feature? The announcement URL goes to the root of the AKS documentation.

The feature is currently available in preview, and documentation is available here: [Node Autoprovisioning NAP](https://azure.microsoft.com/en-us/updates/public-preview-node-autoprovision-support-in-aks/)

**Comment by bplasmeijer** [comment]
> @
> 
> > @stevehipwell Did you find the documentation for this feature? The announcement URL goes to the root of the AKS documentation.
> 
> The feature is currently available in preview, and documentation is available here: [Node Autoprovisioning NAP](https://azure.microsoft.com/en-us/updates/public-preview-node-autoprovision-support-in-aks/)

no windows support.

**Comment by EppO** [comment]
> The only network configuration allowed is [Azure CNI Overlay](https://learn.microsoft.com/en-gb/azure/aks/concepts-network-azure-cni-overlay) with [Powered by Cilium](https://learn.microsoft.com/en-gb/azure/aks/azure-cni-powered-by-cilium).

@justindavies : This is a strong requirement, any chances to cover more AKS network configurations before NAP reaches GA.

**Comment by justindavies** [comment]
@EppO yes, we will be support Azure CNI, CNI Overlay, as well as Cillium at GA

**Comment by AKAL4nka** [comment]
Any news when GA will be out? Before end of year?

**Comment by wdarko1** [comment]
> Any news when GA will be out? Before end of year?

@AKAL4nka Yes, we are targeting GA before the end of the year.

**Comment by wdarko1** [comment]
NAP GA target has been updated to early 2025, currently targeting January 2025.

**Comment by gangster** [comment]
Hello!  Just checking into see if the target is still January 2025, or has that changed?

**Comment by vainkop** [comment]
Any updates?

**Comment by wdarko1** [comment]
Hello! We have updated the GA target to Q1 2025

**Comment by Dgotlieb** [comment]
@wdarko1 do we have any updates as we are approaching the end of Q1 2025?

**Comment by pauska** [comment]
Would be nice with an update on GA release target, since we're now in Q2

**Comment by khamidziyo** [comment]
any updates about GA?

**Comment by haseeb-kp** [comment]
Is there any update on the GA release?

**Comment by audunsolemdal** [comment]
From what I can read Enabling NAP in Auto mode should be supported for Azure CNI without overlay

However I am getting this error:
```
az aks update --name xxx --resource-group yyy --node-provisioning-mode Auto

(OperationNotAllowed) .properties.nodeProvisioningProfile.mode cannot be Auto unless all AgentPools have property .properties.podSubnetID set to one of [''].
Code: OperationNotAllowed
Message: .properties.nodeProvisioningProfile.mode cannot be Auto unless all AgentPools have property .properties.podSubnetID set to one of [''].
```
My pre-existing node pools have different subnet ids set per pool. 

I'm wondering if this a known limitation, and if something will change in this regard before GA?

**Comment by Bryce-Soghigian** [comment]
Pod subnet is not supported

**Comment by wdarko1** [comment]
NAP is GA approved, though we have a few additional experiences (customer logs, a few known [bug fixes](https://github.com/Azure/karpenter-provider-azure/releases/tag/v1.5.0), etc.) that are rolling out over the next few weeks. We expect full readiness of these extra items by end of July 2025.

**Comment by VictorAtPL** [comment]
@wdarko1 hey,

First off, thanks for making Node Auto Provisioning (NAP) GA — it’s a great addition to AKS and already proving really useful.

I’ve got a question about how AKS manages the karpenter-provider-azure codebase for the managed Karpenter/Node Auto Provisioning feature. If I implement a feature in the upstream [Azure/karpenter-provider-azure](https://github.com/Azure/karpenter-provider-azure) repo and it gets merged and released there, what’s the likelihood that it will become available in AKS? Do you maintain your own internal fork and cherry-pick only selected changes, or do you try to stay more or less aligned with the upstream provider on a regular cadence?

The feature I’m considering adding is support for specifying osDiskType (Managed vs Ephemeral) in AKSNodeClass. Right now,  Karpenter for Azure always provisions ephemeral OS disks when the SKU supports them, which on VM sizes with multiple large NVMe drives can cause significant wasted capacity (e.g., ~340 GiB per node on certain Dpds_v6 SKUs). By allowing Managed OS disks, both NVMes could remain fully available for workloads, improving storage utilization and lowering cost for these node types.

**Comment by wdarko1** [comment]
Hi @VictorAtPL, as node auto provisioning (NAP) is based on the open-source [Azure Karpenter Provider](https://github.com/Azure/karpenter-provider-azure) repo, any available features in the Azure Karpenter Provider will be available for NAP in a following AKS release. We encourage any contributions to the open-source provider. That being said, any contributions to the repo will be subject to review/testing to ensure stability prior to merge and release.

**Comment by MShekow** [comment]
Hello, when is supporting **Private** AKS planned, for NAP? Since Private AKS is a typical hardening approach, not being able to use NAP is kind of a weird prioritization on your end.

**Comment by josefallasv** [comment]
@MShekow FYI, NAP now supports private AKS as indicated in https://learn.microsoft.com/en-us/azure/aks/node-autoprovision?tabs=azure-cli:
"Private Clusters are now supported with NAP-enabled clusters, which were previously listed as unsupported."
